kerbauth.go

```diff
@@ -0,0 +1,123 @@
+package mssql
```

Comment:
I agree 1.13 is probably a reasonable minimum requirement at this point.
That said - it may be worthwhile to decouple specific auth implementations like this one from msdsn in general and instead have each auth implementation inject itself into a discovery mechanism such as with some visitor pattern on the connection properties. I'm new enough to Go not to know the best way to have such dependency injection. 
With such dependency injection you could mark the kerbauth files for 1.13+ only.

Reply:
How could this PR align with #718  ?

Reply:
@shueybubbles with 718, anyone using go-mssql driver for kerberos will have to implement their own custom logic and/or maintain their own repos to handle the entire kerberos logic.
Moreover my PR provides with flexibility to use either krb cache or keytab file which I believe is missing from 718?

Reply:
My other comment re: coupling the config to the specific krb implementation is my main concern. The other PR at least avoids such coupling. I think there should be a way for the backend of the driver to manage krb more transparently than having the client use krb-specific packages to even define the connection.